be/src/vec/functions/function_from_unixtime.cpp

```diff
@@ -0,0 +1,301 @@
+            }
+        }
+    }
+};
+
+template <typename from_type, ArgumentNumber argument_num>
+class FunctionFromUnixTime : public IFunction {
+    static_assert(std::is_same_v<from_type, Int64> || std::is_same_v<from_type, Float64> ||
+                          IsDecimalNumber<from_type>,
+                  "from_unixtime only support using Int64, Decimal or Float64 as first arugment.");
+
+public:
+    static constexpr auto name = "from_unixtime";
+    String get_name() const override { return name; }
+
+    static FunctionPtr create() { return std::make_shared<FunctionFromUnixTime>(); }
+
+    size_t get_number_of_arguments() const override {
+        if constexpr (argument_num == ArgumentNumber::Two) {
+            return 2;
+        } else {
+            return 1;
+        }
+    }
+
+    ColumnNumbers get_arguments_that_are_always_constant() const override {
+        if constexpr (argument_num == ArgumentNumber::Two) {
+            return {1};
+        } else {
+            return {};
+        }
+    }
+
+    DataTypes get_variadic_argument_types_impl() const override {
+        if constexpr (IsDecimalNumber<from_type>) {
+            if constexpr (argument_num == ArgumentNumber::Two) {
+                return {std::make_shared<vectorized::DataTypeDecimal<Decimal32>>(9, 0),
+                        std::make_shared<vectorized::DataTypeString>()};
+            } else {
+                return {std::make_shared<vectorized::DataTypeDecimal<Decimal32>>(9, 0)};
+            }
+        } else if constexpr (std::is_same_v<from_type, Int64>) {
+            if constexpr (argument_num == ArgumentNumber::Two) {
+                return {std::make_shared<vectorized::DataTypeInt64>(),
+                        std::make_shared<vectorized::DataTypeString>()};
+            } else {
+                return {std::make_shared<vectorized::DataTypeInt64>()};
+            }
+        } else if constexpr (std::is_same_v<from_type, Float64>) {
+            if constexpr (argument_num == ArgumentNumber::Two) {
+                return {std::make_shared<vectorized::DataTypeFloat64>(),
+                        std::make_shared<vectorized::DataTypeString>()};
+            } else {
+                return {std::make_shared<vectorized::DataTypeFloat64>()};
+            }
+        }
+    }
+
+    DataTypePtr get_return_type_impl(const ColumnsWithTypeAndName& arguments) const override {
+        if (arguments.empty()) {
+            throw doris::Exception(ErrorCode::INVALID_ARGUMENT,
+                                   "from_unixtime() requires at least 1 argument, 0 provided.");
+        }
+
+        if constexpr (argument_num == ArgumentNumber::Two) {
+            if (arguments.size() != 2) {
+                throw doris::Exception(ErrorCode::INTERNAL_ERROR,
+                                       "from_unixtime() requires 2 arguments, {} provided.",
+                                       arguments.size());
+            }
```

Comment:
I think we shouldn't do checks like these. It's part of BE's function match. this is redundant

Reply:
Actually in clickhouse, function signature check is almost alway added.